### PR TITLE
Add MaxGasLimit per transaction

### PR DESCRIPTION
**Pull request by KamiD**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okex/okexchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okex/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: x/evm/types/params.go [hunk 1]
```diff
@@ -130,3 +139,7 @@ func validateEIPs(i interface{}) error {
 
 	return nil
 }
+
+func validateMaxGasLimit(_ interface{}) error {
```
Comment (xiangjianmeng) [inline x/evm/types/params.go]: should check the type of params

Comment (KamiD) [reply x/evm/types/params.go]: fixed

File: x/evm/types/params.go [hunk 2]
```diff
@@ -45,18 +50,20 @@ type Params struct {
 	EnableContractDeploymentWhitelist bool `json:"enable_contract_deployment_whitelist" yaml:"enable_contract_deployment_whitelist"`
 	// EnableContractBlockedList controls the availability of contracts
 	EnableContractBlockedList bool `json:"enable_contract_blocked_list" yaml:"enable_contract_blocked_list"`
+	// MaxGasLimit defines the max gas limit in transaction
+	MaxGasLimitPerTx uint64 `json:"max_gas_limit" yaml:"max_gas_limit"`
```
Comment (xiangjianmeng) [inline x/evm/types/params.go]: `json:"max_gas_limit_per_tx" yaml:"max_gas_limit_per_tx"

File: app/ante/ante.go [hunk 3]
```diff
@@ -56,6 +56,7 @@ func NewAnteHandler(ak auth.AccountKeeper, evmKeeper EVMKeeper, sk types.SupplyK
 			anteHandler = sdk.ChainAnteDecorators(
 				NewEthSetupContextDecorator(), // outermost AnteDecorator. EthSetUpContext must be called first
 				NewEthMempoolFeeDecorator(evmKeeper),
+				NewGasLimitDecorator(evmKeeper),
```
Comment (AaronZgl) [inline app/ante/ante.go:58]: This ante handler can be placed at the top

Comment (KamiD) [reply app/ante/ante.go:58]: done

**Review by xiangjianmeng — APPROVED** [review]

**Review by summerpro — APPROVED** [review]

**Review by AaronZgl — APPROVED** [review]

**Pull request merged**